### ISSUE TITLE
[no-type-alias] `Readonly<[A, B]>` should be allowed same as `readonly [A, B]` by `allowTupleTypes`

### ISSUE THREAD

**Issue by JounQin** [typescript-eslint/typescript-eslint#3181]
- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [read the FAQ](https://github.com/typescript-eslint/typescript-eslint/blob/master/docs/getting-started/linting/FAQ.md) and my problem is not listed.

**Repro**

```JSON
{
  "rules": {
    "@typescript-eslint/no-type-alias": [
      2,
      {
        "allowAliases": "in-unions-and-intersections",
        "allowCallbacks": "always",
        "allowConditionalTypes": "always",
        "allowConstructors": "always",
        "allowLiterals": "in-unions-and-intersections",
        "allowMappedTypes": "always",
        "allowTupleTypes": "always"
      }
    ]
  }
}
```

```TS
import type { JSXElement, JSXFragment } from '@babel/types'

// if tuple types are used as generic, it should be allowed
export type JsxTypes = Readonly<[JSXElement['type'], JSXFragment['type']]>

export type JsxType = JsxTypes[number] // any option to allow this?
```

**Expected Result**

No error

**Actual Result**

not allowed

**Additional Info**

**Versions**

| package                            | version   |
| ---------------------------------- | --------- |
| `@typescript-eslint/eslint-plugin` | `4.17.0`  |
| `@typescript-eslint/parser`        | `4.17.0`  |
| `TypeScript`                       | `4.2.3`   |
| `ESLint`                           | `7.21.0`  |
| `node`                             | `14.16.0` |

**Comment by JounQin** [comment]
I think we can add a new option `allowGeneric` for the first case?

**Comment by JounQin** [comment]
Actually I just want to disable codes like `type A = B`, but it's not easy to set for now.

**Comment by bradzacher** [comment]
in `type A = B`, `B` is a `TSTypeReference` which falls under `allowAliases`.

**Comment by JounQin** [comment]
@bradzacher I mean disallow `type A = B` only but allow any other usages.

**Comment by bradzacher** [comment]
there is no mechanism to disable just type reference aliases.

All of the options and what they apply to are [listed in the docs](https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/no-type-alias.md#options)

You could likely build a selector to match this and block it via [no-restricted-syntax](https://eslint.org/docs/rules/no-restricted-syntax)

**Comment by JounQin** [comment]
Thanks, I'll give that a try.

**Comment by JounQin** [comment]
@bradzacher  Thanks a lot, I finished up by:

```js
{
  'no-restricted-syntax': [
    2,
    {
      selector:
        'TSTypeAliasDeclaration > .typeAnnotation[type=/^(TS.*Keyword|TSTypeReference)$/]',
      message:
        'Type alias references `Ts*Keyword`s or `TSTypeReference` directly is not allowed',
    },
  ],
}
```

Is it worth to add into `no-type-alias`'s documentation?

**Comment by bradzacher** [comment]
This can now be allowed via the option added in #3865
